### ISSUE TITLE
Repo register without provider enrolled gives confusing error

### ISSUE THREAD

**Issue by JAORMX** [stacklok/minder#1982]
### Describe the issue

When running `minder repo register -p github` I get the following error:

```
No config file present, using default values.
Usage:
  minder repo register [flags]

Flags:
  -h, --help          help for register
  -n, --name string   List of repository names to register, i.e owner/repo,owner/repo

Global Flags:
      --config string            Config file (default is $PWD/config.yaml)
      --grpc-host string         Server host (default "staging.stacklok.dev")
      --grpc-insecure            Allow establishing insecure connections
      --grpc-port int            Server port (default 443)
      --identity-client string   Identity server client ID (default "minder-cli")
      --identity-url string      Identity server issuer URL (default "https://auth.staging.stacklok.dev")
  -j, --project string           ID of the project
  -p, --provider string          Name of the provider, i.e. github (default "github")

Message: Error getting list of remote repos
Details: Permission denied
```

This is because I did not have the provider enrolled. The error should relate to the
provider not being enrolled, and not to the permissions I have.

### To Reproduce

In an environment without a provider enrolled do:
* `minder auth login`
* `minder repo register -p github`

### What version are you using?

`main`

**Comment by eleftherias** [comment]
@JAORMX I thought this was fixed with #1515.

I tried recently registering a repo without enrolling and got this error:
```
Usage:
  minder repo register [flags]

Flags:
  -h, --help          help for register
  -n, --name string   List of repository names to register, i.e owner/repo,owner/repo

Global Flags:
      --config string            Config file (default is $PWD/config.yaml)
      --grpc-host string         Server host (default "staging.stacklok.dev")
      --grpc-insecure            Allow establishing insecure connections
      --grpc-port int            Server port (default 443)
      --identity-client string   Identity server client ID (default "minder-cli")
      --identity-url string      Identity server issuer URL (default "https://auth.staging.stacklok.dev")
  -j, --project string           ID of the project
  -p, --provider string          Name of the provider, i.e. github (default "github")

Message: Error getting list of remote repos
Details: cannot get access token for provider: did you run `minder provider enroll`?
```

**Comment by rdimitrov** [comment]
I've tried the reproduce steps in the issue just now with the latest main and confirm the behaviour @eleftherias posted above. Closing the issue as fixed 👍
